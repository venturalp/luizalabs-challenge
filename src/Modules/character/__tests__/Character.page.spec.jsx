import { screen, waitFor } from '@testing-library/dom'
import { renderWithRouter, ThemeWrapper } from 'Commons/tests/Tests.Helpers'
import { HomePage } from 'Modules/home/Home.page'
import React from 'react'

describe('test home page', () => {
  it('test check favorites', async () => {
    const { asFragment } = renderWithRouter(
      <ThemeWrapper>
        <HomePage />
      </ThemeWrapper>,
    )

    let favorites
    await waitFor(
      async () => (favorites = await screen.queryAllByTestId('toggleFavorite')),
    )
    expect(asFragment()).toMatchSnapshot()
    console.log(favorites)
  })
})
