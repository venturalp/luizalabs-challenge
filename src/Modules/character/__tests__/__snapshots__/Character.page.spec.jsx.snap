// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home page test check favorites 1`] = `
<DocumentFragment>
  <div
    class="sc-iBPRYJ kEDcat"
  >
    <img
      class="sc-bdfBwQ dJiJvq"
      src="logo2x.png"
    />
    <form
      class="sc-fubCfw cMcwNv"
    >
      <h1>
        EXPLORE O UNIVERSO
      </h1>
      <p>
        Mergulhe no domínio deslumbrante de todos os personagens clássicos que você ama - e aqueles que você descobrirá em breve!
      </p>
      <input
        autocomplete="off"
        class="sc-gsTCUz ecbFyq"
        name="searchBar"
        placeholder="Procure por heróis"
        type="text"
        value=""
      />
    </form>
    <div
      class="sc-jSgupP faYPOp sc-gKsewC ifUxxl sc-pFZIQ eQITDV"
    >
      <p>
        Encontrado undefined herói
      </p>
      <div
        class="sc-jSgupP faYPOp"
      >
        <div
          class="orderName"
        >
          <img
            alt="hero"
            src="hero.png"
          />
          <p>
            Ordenar por nome - A/Z
          </p>
          <span
            class="sc-dlfnbm jRrDak"
          >
            <img
              alt="toggleIco"
              src="ToggleOn.png"
            />
          </span>
        </div>
        <div
          class="favorites"
        >
          <span
            class="sc-dlfnbm jRrDak"
          >
            <img
              alt="toggleIco"
              src="heartEmpty.png"
            />
            <p>
              Somente favoritos
            </p>
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-kEjbxe khZWnm"
    />
    <footer
      class="sc-jrAGrp kAkGro"
    />
  </div>
</DocumentFragment>
`;
